File: Scripts/server/palindrome-detection/c/palindrome-detection.c
#include <string.h>

int palindrome(const char*);

int executeTask(int i) {
	char* array = "saippuakivikauppias";
	palindrome(array);

   return i + 1;
}

int main() {

   	volatile int r = 1;
	for (int i=0; i<2000000000; ++i) {
		r = executeTask(i + r);
	}
	return 0;
}

int palindrome(const char *s)
{
   int l;
   l = strlen(s);
   for(int i=0; i<l/2; i++)
   {
     if ( s[i] != s[l-i-1] ) return 0;
   }
   return 1;
}

File: scripts/task_tested_with_compile_script/palindrome-detection/c-sharp/palindrome-detection.cs
using System;

class Program
{
    static string Reverse(string value)
    {
        char[] chars = value.ToCharArray();
        Array.Reverse(chars);
        return new string(chars);
    }
 
    static bool IsPalindrome(string value)
    {
        return value == Reverse(value);
    }
 
    private static volatile int r = 0;

    public static int executeTask(int i) {

	IsPalindrome("saippuakivikauppias" + i);
	return i;
    }	

    static void Main(string[] args)
    {
	for (int i = 0; i < 2000000000; ++i)
    		r = executeTask(i);
    }
}
